Include/internal/pycore_typeobject.h

```diff
@@ -61,6 +61,8 @@ typedef struct {
     PyObject *tp_weaklist;
 } static_builtin_state;
 
+#define TYPE_VERSION_CACHE_SIZE (1<<12)  /* Must be a power of 2 */
```

Comment:
Not a big deal, but have we experimented with different values here? Might be useful to try out different values once this lands.

Reply:
Yeah I haven't... That would make sense. I also wonder about adding the possibility of adding some diagnostics metadata for cache misses?